src/dog/services/dog/dog.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable } from '@nestjs/common';
+import { DOGS } from 'src/dog/datasource/dogs';
+
+@Injectable()
+export class DogService {
+  find(dogId: number) {
+    let d: any;
+    DOGS.forEach((dog) => {
+      if (dog.id == dogId) {
+        d = dog;
+      }
+    });
+    return d;
+  }
+  findOne(age: number, race: string) {
+    let d: any;
+    DOGS.forEach((dog) => {
+      if (dog.age == age && dog.race == race) {
+        d = dog;
+      }
+    });
+    return d;
+  }
```

Comment:
Same comment as in the find function. Please improve your search to be more performant.